cpp/src/replace/replace.cu

```diff
@@ -13,97 +31,571 @@
+
+namespace cudf {
+namespace detail {
+std::unique_ptr<cudf::column> find_and_replace_all(cudf::column_view const& input_col,
+                                                   cudf::column_view const& values_to_replace,
+                                                   cudf::column_view const& replacement_values,
+                                                   rmm::mr::device_memory_resource* mr,
+                                                   cudaStream_t stream) {
+  if (0 == input_col.size() || 0 == values_to_replace.size() || 0 == replacement_values.size())
+      {
+    return std::make_unique < cudf::column > (input_col);
+  }
+
+  CUDF_EXPECTS(values_to_replace.size() == replacement_values.size(),
+               "values_to_replace and replacement_values size mismatch.");
+
+  if (0 == values_to_replace.size())
+      {
+    return std::make_unique < cudf::column > (input_col);
+  }
+
+  CUDF_EXPECTS(input_col.type() == values_to_replace.type() &&
+                   input_col.type() == replacement_values.type(),
+               "Columns type mismatch");
+  CUDF_EXPECTS(values_to_replace.nullable() == false,
+               "values_to_replace must be non-nullable");
+
+  return cudf::experimental::type_dispatcher(input_col.type(),
+                                             replace_kernel_forwarder { },
+                                             input_col,
+                                             values_to_replace,
+                                             replacement_values,
+                                             mr,
+                                             stream);
+}
+
+} //end details
+
+namespace experimental {
+/* --------------------------------------------------------------------------*/
+/*
+ * @brief Replace elements from `input_col` according to the mapping `values_to_replace` to
+ *        `replacement_values`, that is, replace all `values_to_replace[i]` present in `input_col`
+ *        with `replacement_values[i]`.
+ *
+ * @param[in] col gdf_column with the data to be modified
+ * @param[in] values_to_replace gdf_column with the old values to be replaced
+ * @param[in] replacement_values gdf_column with the new values
+ *
+ * @returns output gdf_column with the modified data
+ */
+/* ----------------------------------------------------------------------------*/
+std::unique_ptr<cudf::column> find_and_replace_all(cudf::column_view const& input_col,
+                                                   cudf::column_view const& values_to_replace,
+                                                   cudf::column_view const& replacement_values,
+                                                   rmm::mr::device_memory_resource* mr) {
+  return cudf::detail::find_and_replace_all(input_col, values_to_replace, replacement_values, mr, 0);
+}
+} //end experimental
+} //end cudf
+
+namespace { //anonymous
+
+template<typename Type, typename iter>
+__global__
+void replace_nulls(cudf::size_type size,
```

Comment:
This kernel could be nicely replaced with a simple `thrust::transform_if`. https://thrust.github.io/doc/group__transformations_gaaf162576bd1daef87b80ea82bfbf8558.html

Reply:
Done.